app/main.py

```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+from typing import Callable, Any
+from functools import wraps
+
+
+def _int_float_types_execption(func: Callable) -> Callable:
+    @wraps(func)
+    def inner(self: Any, other: Any) -> Callable:
+        if type(other) in (int, float):
+            return func(self, other=Distance(other))
+        else:
+            return func(self, other)
```

Comment:
```suggestion
    def inner(self: Any, other: Any) -> Callable:
        if isinstance(other, (int, float)):
            return func(self, other=Distance(other))
        return func(self, other)
```
also better use `isinstance`

Reply:
Ahh, I understand what you mean.